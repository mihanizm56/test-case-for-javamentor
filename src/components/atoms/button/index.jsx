import React, { memo } from "react";
import ButtonMaterial from "@material-ui/core/Button";

export const Button = memo(({ handleClick }) => (
  <ButtonMaterial onClick={handleClick} to="/profile">
    Выбрать
  </ButtonMaterial>
));
