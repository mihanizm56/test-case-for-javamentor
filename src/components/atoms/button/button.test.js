import React from 'react';
import {shallow} from 'enzyme';
import {Button} from '../button';

describe('test FooterView', () => {
    const buttonProps = {
        handleClick:jest.fn()
    }
    let component

    beforeEach(() => {
        jest.clearAllMocks();
        component = shallow(<Button {...buttonProps}/>);
    });

    describe('test render contract', () => {
        it('renders inside html-content', () => {
            expect(component.dive().dive()).toMatchSnapshot();
        })

        it('renders Button', () => {
            expect(component.dive().dive().find('Button').length).toBe(1);
        })
    })

    describe('test props contract', () => {
        it('check handleClick prop', () => {
            expect(component.dive().dive().find('Button').prop('onClick')).toEqual(buttonProps.handleClick);
        })
    })
})
    