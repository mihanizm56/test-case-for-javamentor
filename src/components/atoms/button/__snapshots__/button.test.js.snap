// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FooterView test render contract renders inside html-content 1`] = `
<Button
  className="bootstrap-button"
  classes={
    Object {
      "colorInherit": "MuiButton-colorInherit-23",
      "contained": "MuiButton-contained-13",
      "containedPrimary": "MuiButton-containedPrimary-14",
      "containedSecondary": "MuiButton-containedSecondary-15",
      "disabled": "MuiButton-disabled-22",
      "extendedFab": "MuiButton-extendedFab-20",
      "fab": "MuiButton-fab-19",
      "flat": "MuiButton-flat-7",
      "flatPrimary": "MuiButton-flatPrimary-8",
      "flatSecondary": "MuiButton-flatSecondary-9",
      "focusVisible": "MuiButton-focusVisible-21",
      "fullWidth": "MuiButton-fullWidth-27",
      "label": "MuiButton-label-3",
      "mini": "MuiButton-mini-24",
      "outlined": "MuiButton-outlined-10",
      "outlinedPrimary": "MuiButton-outlinedPrimary-11",
      "outlinedSecondary": "MuiButton-outlinedSecondary-12",
      "raised": "MuiButton-raised-16",
      "raisedPrimary": "MuiButton-raisedPrimary-17",
      "raisedSecondary": "MuiButton-raisedSecondary-18",
      "root": "MuiButton-root-2 WithStyles-Button--root-1",
      "sizeLarge": "MuiButton-sizeLarge-26",
      "sizeSmall": "MuiButton-sizeSmall-25",
      "text": "MuiButton-text-4",
      "textPrimary": "MuiButton-textPrimary-5",
      "textSecondary": "MuiButton-textSecondary-6",
    }
  }
  color="default"
  component="button"
  disableFocusRipple={false}
  disabled={false}
  fullWidth={false}
  mini={false}
  onClick={[MockFunction]}
  size="medium"
  type="button"
  variant="text"
>
  Загрузить
</Button>
`;
